proposals/0064-transaction-receipt.md

```diff
@@ -0,0 +1,298 @@
+constructed once per slot.
+
+### Design Goals
+
+1. Transaction Receipts should be deterministic.
+   Given a transaction T and ledger history leading up to it, serializing the
+   receipt generated for T  should result in the same byte vector for all nodes
+   in the network.
+   *Rationale:* Determinism is required for cluster-wide consensus.
+
+2. Transaction Receipts should not be required during block construction.
+   *Rationale:* Future upgrades propose tolerating asynchronous replay during
+   block construction. In other words, validators should be allowed to produce
+   and distribute a block before replaying said block. It is impossible to
+   introduce such a tolerance if transaction receipts are mandatory components
+   of blocks.
+
+## Alternatives Considered
+
+### Using TransactionStatusMeta
+
+An alternative to introducing a new transation receipt type is reusing the transaction
+status data as it appears in RPC ([TransactionStatusMeta]).  This would reduce
+complexity for clients.
+
+However, *TransactionStatusMeta* has no strict definition and is thus malleable
+in violation of design goal 1. Technical reasons are as follows:
+
+- Available fields vary between node releases.
+- Log data is truncated based on node configuration.
+- The [TransactionResult] type includes error codes, which are
+  implementation-defined (Thus breaks multi-client compatibility).
+
+This would make a commitment scheme impractical as-is. Addressing these concerns
+is a breaking change.
+
+  [TransactionStatusMeta]: https://docs.rs/solana-transaction-status/1.16.1/solana_transaction_status/struct.TransactionStatusMeta.html
+  [TransactionResult]: https://docs.rs/solana-sdk/1.16.1/solana_sdk/transaction/type.Result.html
+
+### Proof-of-History / Block Hash
+
+Another alternative to introducing a new commitment scheme is reusing the
+proof-of-history (PoH) hash chain. This was proposed in
+[SPV Proposal](https://docs.solana.com/proposals/simple-payment-and-state-verification#transaction-merkle).
+
+The PoH chain currently commits to the signatures of all transactions added to
+the ledger. The consensus layer then periodically votes on the last state of the
+PoH chain for each bankhash which is an extension of the blockhash.
+Expanding the PoH hash is the least complex option as of today but
+is consequential for future upgrades.
+
+Such a change would be incompatible with design goal 2 because it redefines
+the PoH hash to additionally commit to execution results, instead of only
+ledger content. Block producers are then forced to synchronously replay while
+appending PoH ticks.
+
+Furthermore, it significantly changes behavior in the event of an execution
+disagreement (e.g. due to a difference in execution behavior between
+validators). Mixing execution results into PoH forces execution disagreements to
+result in a chain split.
+
+## Detailed Design
+
+### TransactionReceiptData Specification
```

Comment:
Who and where are these messages propagated by?

Reply:
This is not entirely in scope for this SIMD, but that's mentioned in #52 and we've referenced it in the impact section in this one.